inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp

```diff
@@ -818,8 +652,87 @@ struct typed_resizeLinearF32C1 {
                   const float alpha[], const int mapsx[],
                   const float beta[], const Size& inSz,
                   const Size& outSz, const int lpi, const int length) {
-                calcRowLinear32FC1Impl<Mapper>(isa_tag_t{}, dst, src0, src1, alpha,
-                                               mapsx, beta, inSz, outSz, lpi, length);
+            calcRowLinear32FC1Impl<Mapper>(isa_tag_t{}, dst, src0, src1, alpha,
+                                           mapsx, beta, inSz, outSz, lpi, length);
+        };
+    }
+};
+}  // namespace
+
+namespace {
+
+using resizeLinearU8C3C4_suptypes = typelist<uint8_t>;
+
+template<class Mapper, int chs>
+inline void calcRowLinear8UC3C4Impl(scalar_tag,
+                                    std::array<std::array<uint8_t*, 4>, chs>& dst,
+                                    const uint8_t* src0[],
+                                    const uint8_t* src1[],
+                                    const short    alpha[],
+                                    const short    clone[],  // 4 clones of alpha
+                                    const short    mapsx[],
+                                    const short    beta[],
+                                    uint8_t        tmp[],
+                                    const Size&    inSz,
+                                    const Size&    outSz,
+                                    const int      lpi,
+                                    const int      length) {
+    using alpha_type = typename Mapper::alpha_type;
+    for (int l = 0; l < lpi; l++) {
+        constexpr static const auto unity = Mapper::unity;
+
+        auto beta0 = beta[l];
+        auto beta1 = saturate_cast<alpha_type>(unity - beta[l]);
+
+        for (int x = 0; x < length; x++) {
+            auto alpha0 = alpha[x];
+            auto alpha1 = saturate_cast<alpha_type>(unity - alpha[x]);
+            auto sx0 = mapsx[x];
+            auto sx1 = sx0 + 1;
+
+            for (int c = 0; c < chs; c++) {
```

Comment:
`chs` seems constexpr value ( template declaration) and it looks like we can make loop-unrolling for this for

but perhaps c++ compiler was able to do that...


Anyway - if we rearrange several cycles here in different order (dynamic loop, based on `length` is first, then by `lpi` and constexp `chs`)  I believe we can increase performance here: to prevent cache push-out for `alfa`, `mapsx` on every outer iteration

Of course is this part means performance-critical and we have free time for experiments:)

Reply:
Thank you for useful comment. I'll try to unroll this loop for performance improvement in the separated task.